src/daft-image/src/image_buffer.rs

```diff
@@ -126,27 +126,26 @@ impl<'a> DaftImageBuffer<'a> {
         )
         .map_err(|e| {
             DaftError::ValueError(format!(
-                "Encoding image into file format {} failed: {}",
-                image_format, e
+                "Encoding image into file format {image_format} failed: {e}"
             ))
         })
     }
 
     pub fn fit_to(&self, w: u32, h: u32) -> Self {
         // Preserving aspect ratio, resize an image to fit within the specified dimensions.
         let scale_factor = {
-            let width_scale = w as f64 / self.width() as f64;
-            let height_scale = h as f64 / self.height() as f64;
+            let width_scale = f64::from(w) / f64::from(self.width());
```

Comment:
not a big fan of this rule as I think `w as f64` is definitely more common and easier to write than `f64::from(w)`

Reply:
The lint suggesting the use of the `From` trait instead of the `as` operator for type conversion is useful, despite the `as` operator being more common and seemingly easier to write.

1. **Type Safety**: The `From` trait provides stronger type safety guarantees.
   - `From::from()` ensures type compatibility at compile-time.

2. **Explicit Error Handling**: 
   - When using `TryFrom` (for fallible conversions), you're forced to handle potential conversion failures explicitly.
   - The `as` operator may silently truncate or wrap values, masking potential issues.

Reply:
696f1f18173a0aa3f126da67aa61cc560ecd389b